web/packages/teleterm/src/ui/DocumentTerminal/DocumentTerminal.tsx

```diff
@@ -118,9 +117,16 @@ export function DocumentTerminal(props: {
       autoFocusDisabled={true}
     >
       {$fileTransfer}
-      {ptyProcess && (
+      {attempt.status === 'success' && (
         <Terminal
-          ptyProcess={ptyProcess}
+          // The key prop makes sure that we render Terminal only once for each PTY process.
+          //
+          // In a scenario where startError has occured, when the user initializes the PTY process
+          // again, we want to re-run all the hooks in Terminal for this new PTY process.
+          key={attempt.data.ptyProcess.getPtyId()}
```

Comment:
This follows the approach explained in [Resetting all state when a prop changes](
https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes) from You Might Not Need an Effect.